### PR TITLE
354 approve project

### PR DESCRIPTION
**Pull request by gjermundgaraba**

### CODE REVIEW

**Comment by gjermundgaraba** [comment]
Forgot to use the status to not allow issuing credits to a non-approved project

File: chain/x/plasticcredit/keeper/project.go [hunk 1]
```diff
@@ -57,6 +58,39 @@ func (k Keeper) CreateProject(ctx sdk.Context, creator sdk.AccAddress, applicant
 	return nextID, nil
 }
 
+func (k Keeper) ApproveProject(ctx sdk.Context, approver sdk.AccAddress, projectID uint64) error {
+	project, found := k.GetProject(ctx, projectID)
+	if !found {
+		return errors.Wrapf(plasticcredit.ErrNotFoundProject, "project with id %d was not found", projectID)
+	}
+
+	// At some point, I would like to have some better indexing that would allow us to not have to fetch so many things just to get to the issuer
```
Comment (ba1ciu) [inline chain/x/plasticcredit/keeper/project.go:67]: Yeah, might be worth to include additional indexes on the "many" side of the relation or do you have something else in mind?

File: chain/x/plasticcredit/errors.go [hunk 2]
```diff
@@ -35,8 +35,9 @@ var (
 	ErrNotFoundCreditClass          = errors.Register(ModuleName, 4404, "credit class not found")
 	ErrDuplicateCreditClass         = errors.Register(ModuleName, 4409, "duplicate credit class detected")
 
-	ErrNotFoundProject  = errors.Register(ModuleName, 5404, "project not found")
-	ErrDuplicateProject = errors.Register(ModuleName, 5409, "duplicate project detected")
+	ErrProjectNotApproved = errors.Register(ModuleName, 5001, "project is not approved")
```
[the remark this answers is not in the document]
Comment (gjermundgaraba) [reply chain/x/plasticcredit/errors.go:38]: Yes, I totally agree. Let's make a decision on that and create a separate task to clean it up

File: chain/x/plasticcredit/keeper/msg_server_test.go [hunk 3]
```diff
@@ -143,8 +143,8 @@ func (s *TestSuite) TestCreateIssuer() {
 					Admin:       tc.msg.Admin,
 				}, issuer)
 
-				s.Require().Len(events, 2)
-				parsedEvent, err := sdk.ParseTypedEvent(events[1])
+				s.Require().Len(events, 1)
```
[the remark this answers is not in the document]
Comment (gjermundgaraba) [reply chain/x/plasticcredit/keeper/msg_server_test.go:146]: I have done quite a bit about it in this PR actually. In the populate function I reset the events, so hopefully less updates in the future :) We'll see!

File: chain/x/plasticcredit/keeper/project.go (hunk 1 above)
Comment (gjermundgaraba) [reply chain/x/plasticcredit/keeper/project.go:67]: Not sure. I just put down the note for later, but it looks like for instance the ORM has builtin support to make this easier. Something I would like to understand better later on.

**Pull request merged**